src/python_testing/TC_CC_10_1.py

```diff
@@ -0,0 +1,484 @@
+            await asyncio.sleep(10)
+
+        self.step("2g")
+        if self.pics_guard(self.check_pics("CC.S.C0a.Rsp")):
+            result = await self.TH1.ReadAttribute(self.dut_node_id, [(1, attributes.ColorTemperatureMireds)])
+            asserts.assert_less_equal(result[1][cluster][attributes.ColorTemperatureMireds], ColorTempPhysicalMaxMiredsValue,
+                                      "ColorTemperatureMireds is not less than or equal to ColorTempPhysicalMaxMireds")
+            asserts.assert_greater_equal(result[1][cluster][attributes.ColorTemperatureMireds], ColorTempPhysicalMinMiredsValue,
+                                         "ColorTemperatureMireds is not greater than or equal to ColorTempPhysicalMinMireds")
+
+        self.step("2h")
+        if self.pics_guard(self.check_pics("CC.S.C43.Rsp")):
+            await self.TH1.SendCommand(self.dut_node_id, 1, cluster.Commands.EnhancedMoveToHueAndSaturation(20000, 50, 0, 1, 1))
+
+        self.step("2i")
+        if self.pics_guard(self.check_pics("CC.S.C43.Rsp")):
+            result = await self.TH1.ReadAttribute(self.dut_node_id, [(1, attributes.EnhancedCurrentHue), (1, attributes.CurrentSaturation)])
+            asserts.assert_less_equal(result[1][cluster][attributes.EnhancedCurrentHue], 21800,
+                                      "EnhancedCurrentHue is not less than or equal to 21800")
+            asserts.assert_greater_equal(result[1][cluster][attributes.EnhancedCurrentHue], 18200,
+                                         "EnhancedCurrentHue is not greater than or equal to 18200")
+            asserts.assert_less_equal(result[1][cluster][attributes.CurrentSaturation], 58, "CurrentSaturation is not 58")
+            asserts.assert_greater_equal(result[1][cluster][attributes.CurrentSaturation], 42, "CurrentSaturation is not 42")
+
+        self.step("3")
+        result = await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.ScenesManagement.Commands.StoreScene(self.kGroup1, 0x01))
+        asserts.assert_equal(result.status, Status.Success, "Store Scene failed on status")
+        asserts.assert_equal(result.groupID, self.kGroup1, "Store Scene failed on groupID")
+        asserts.assert_equal(result.sceneID, 0x01, "Store Scene failed on sceneID")
+
+        self.step("4")
+        result = await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.ScenesManagement.Commands.ViewScene(self.kGroup1, 0x01))
+        asserts.assert_equal(result.status, Status.Success, "View Scene failed on status")
+        asserts.assert_equal(result.groupID, self.kGroup1, "View Scene failed on groupID")
+        asserts.assert_equal(result.sceneID, 0x01, "View Scene failed on sceneID")
+        asserts.assert_equal(result.transitionTime, 0, "View Scene failed on transitionTime")
+
+        for EFS in result.extensionFieldSets:
+            if EFS.clusterID == 0x0300:
```

Comment:
It's a bit hard to parse in a diff, but I think you can unnest this a bit by doing 
if EFS.clusterID != 0x0300:
  continue